src/services/configService.test.ts

```diff
@@ -170,121 +151,142 @@ describe("Config Service", () => {
-        language: SupportedRuntimeLanguage.PYTHON,
-        version: "3.6"
-      }
-      expect((sls.service as any as ServerlessAzureConfig).provider.functionRuntime).toEqual(expectedRuntime);
+    describe("Runtime version", () => {
+      it("throws error when unsupported python version in defined", () => {
+        const sls = MockFactory.createTestServerless();
+        sls.service.provider.runtime = "python2.7" as any;
+        expect(() => new ConfigService(sls, {} as any))
+          .toThrowError("Runtime python2.7 is not supported.\n" + 
+            "Runtimes supported: nodejs10,nodejs12,python3.6,python3.7,python3.8");
+        expect((sls.service as any as ServerlessAzureConfig).provider.functionRuntime).toBeUndefined();
+      });
+    
+      it("throws error when incomplete nodejs version in defined", () => {
+        const sls = MockFactory.createTestServerless();
+        sls.service.provider.runtime = "nodejs" as any;
+        expect(() => new ConfigService(sls, {} as any))
+          .toThrowError("Runtime nodejs is not supported.\n" + 
+            "Runtimes supported: nodejs10,nodejs12,python3.6,python3.7,python3.8");
+        expect((sls.service as any as ServerlessAzureConfig).provider.functionRuntime).toBeUndefined();
+      });
+    
+      it("throws error when unsupported nodejs version in defined", () => {
+        const sls = MockFactory.createTestServerless();
+        sls.service.provider.runtime = "nodejs5.x" as any;
+        expect(() => new ConfigService(sls, {} as any))
+          .toThrowError("Runtime nodejs5.x is not supported.\n" + 
+            "Runtimes supported: nodejs10,nodejs12,python3.6,python3.7,python3.8");
+        expect((sls.service as any as ServerlessAzureConfig).provider.functionRuntime).toBeUndefined();
+      });
+    
+      it("Does not throw an error when valid nodejs version is defined", () => {
+        const sls = MockFactory.createTestServerless();
+        sls.service.provider.runtime = Runtime.NODE10
+        expect(() => new ConfigService(sls, {} as any)).not.toThrow();
+        const expectedRuntime: FunctionRuntime = {
+          language: SupportedRuntimeLanguage.NODE,
+          version: "10"
+        }
+        expect((sls.service as any as ServerlessAzureConfig).provider.functionRuntime).toEqual(expectedRuntime);
+      });
+    
+      it("Does not throw an error when nodejs version is defined", () => {
+        const sls = MockFactory.createTestServerless();
+        sls.service.provider.runtime = Runtime.NODE10
+        expect(() => new ConfigService(sls, {} as any)).not.toThrow();
+        const expectedRuntime: FunctionRuntime = {
+          language: SupportedRuntimeLanguage.NODE,
+          version: "10"
+        }
+        expect((sls.service as any as ServerlessAzureConfig).provider.functionRuntime).toEqual(expectedRuntime);
+      });
+    
+      it("Does not throw an error when specific nodejs version is defined", () => {
```

Comment:
is this test still relevant with the new changes?

Reply:
Nope. Removed :)

Reply:
Well, it will be when I push :)